manof/utils/__init__.py

```diff
@@ -55,32 +60,28 @@ def err(self):
         return self._err
 
 
-def git_pull(logger, path):
+async def git_pull(logger, path):
     logger.debug('Pulling', **locals())
-    return shell_run(logger, 'git pull', cwd=path)
+    return await shell_run(logger, 'git pull', cwd=path)
 
 
-@defer.inlineCallbacks
-def shell_run(logger, command, cwd=None, quiet=False, env=None):
+async def shell_run(logger, command, cwd=None, quiet=False, env=None):
```

Comment:
I do not see the benefit of wrapping twisted semantics with asyncio, wrapped again in twisted semantics and doing the inter-conversions back and forth.
The internal method used to execute a subprocesses is `getProcessOutputAndValue` (uses `_callProtocolWithDeferred` which returns a deferred, and is twisted native)
Here you're wrapping deferred in asyncs+awaits, and later up the chain you have to `ensureDeferred()` on that future again.
Is there a necessity or benefit for this mix? 
If not, let's not break the semantics and mix them - Maybe after the py2->py3 conversion manof can ditch twisted entirely for asyncio, entirely